libraries/engage/locations/components/StoreDetailsContent/components/StoresNearbyListItem.jsx

```diff
@@ -0,0 +1,149 @@
+import React, { useContext } from 'react';
+import { useDispatch } from 'react-redux';
+import PropTypes from 'prop-types';
+import { css } from 'glamor';
+import { Button } from '@shopgate/engage/components';
+import { i18n } from '@shopgate/engage/core/helpers';
+import { historyPush } from '@shopgate/engage/core';
+import formatDistance from '../../../helpers/formatDistance';
+import { STORE_DETAILS_PATH } from '../../../constants';
+import { StoreDetailsContext } from '../../../providers/StoreDetailsContext';
+
+const styles = {
+  locationRow: css({
+    borderBottom: '1px solid #e8e8e8',
+    borderTop: '1px solid #e8e8e8',
+  }),
+  distance: css({
+    textWrapMode: 'nowrap',
+    alignContent: 'center',
+    verticalAlign: 'middle',
+    fontWeight: '500',
+  }),
+  makeMyStore: css({
+    textWrapMode: 'nowrap',
+    alignContent: 'center',
+    verticalAlign: 'middle',
+  }),
+
+  makeMyStoreButtonText: css({
+    color: 'var(--color-primary)',
+  }),
+  storeInfo: css({
+    textWrapMode: 'nowrap',
+    alignContent: 'center',
+    verticalAlign: 'middle',
+  }),
+  storeInfoButtonText: css({
+    color: 'var(--color-primary)',
+  }),
+  name: css({
+    verticalAlign: 'middle',
+    alignContent: 'center',
+  }),
+  cellContainer: css({
+    padding: '8px',
+    textAlign: 'end',
+  }),
+  cell: css({
+    verticalAlign: 'middle',
+  }),
+  buttonContainer: css({
+    display: 'flex',
+    gap: '4px 16px',
+    flexWrap: 'wrap',
+    justifyContent: 'flex-end',
+  }),
+  comingSoon: css({
+  }),
+};
+
+/**
+* Shows a location in a row
+* @param {Object} props Props
+* @param {Object} props.location Location
+* @returns {JSX}
+*/
+const StoresNearbyListItem = ({ location }) => {
+  const dispatch = useDispatch();
+  const { preferredLocation, selectLocation } = useContext(StoreDetailsContext);
+  const {
+    name, distance, unitSystem, code, isComingSoon,
+  } = location;
+
+  const isPreferredLocation = preferredLocation && preferredLocation.code === code;
+
+  /**
+   * Opens the store details page
+   * @param {string} locationCode Location code
+   */
+  const openStoreDetails = (locationCode) => {
+    dispatch(historyPush({ pathname: `${STORE_DETAILS_PATH}/${locationCode}` }));
+    // dispatch({
+    //  type: 'PUSH',
+    //  route: {
+    //    pathname: `${STORE_DETAILS_PATH}/${locationCode}`,
+    //  },
+    // });
+  };
+
+  return (
+    <tr className={styles.locationRow}>
+      <td className={styles.cell}>
+        <div className={styles.cellContainer}>
```

Comment:
the nearby locations are now right aligned. is this on purpose?

![Bildschirmfoto 2025-01-27 um 12 39 21](https://github.com/user-attachments/assets/050f2b5a-f3ce-4863-b457-2ecea2128bd5)

maybe you can tweak the spacing a bit. on an iPhone 14 Max there is a lot of unused whitespace around the distance row.

Reply:
done